src/screens/WhotoFollow/index.js

```diff
@@ -177,65 +177,70 @@ const WhotoFollow = () => {
-            setProfileId({
-              anonProfileId: anonymousUserId,
-              signedProfileId: userId,
-              token: res.token,
-              anonymousToken: res.anonymousToken
+      registerUser(data)
+        .then(async (res) => {
+          setFetchRegister(false);
+          if (res.code === 200) {
+            TokenStorage.set(res);
+            setToken(res.token);
+            try {
+              const userId = await JwtDecode(res.token).user_id;
+              const anonymousUserId = await JwtDecode(res.anonymousToken).user_id;
+              setProfileId({
+                anonProfileId: anonymousUserId,
+                signedProfileId: userId,
+                token: res.token,
+                anonymousToken: res.anonymousToken
+              });
+            } catch (e) {
+              crashlytics().recordError(new Error(e));
+            }
+            showMessage({
+              message: 'Welcome to Better Social',
+              type: 'success',
+              backgroundColor: colors.holytosca
+            });
+            setTimeout(() => {
+              create();
+              setImage(null, usersDispatch);
+              setInitialValue({id: res.token});
+            }, 2000);
+          } else {
+            crashlytics().recordError(new Error(res));
+            showMessage({
+              message: 'Cannot connect to server, please try again later',
+              type: 'danger',
+              backgroundColor: colors.red
             });
-          } catch (e) {
-            crashlytics().recordError(new Error(e));
           }
-          showMessage({
-            message: 'Welcome to Better Social',
-            type: 'success',
-            backgroundColor: colors.holytosca
-          });
-          setTimeout(() => {
-            create();
-            setImage(null, usersDispatch);
-            setInitialValue({id: res.token});
-          }, 2000);
-        } else {
-          crashlytics().recordError(new Error(res));
+        })
+        .catch((error) => {
+          crashlytics().recordError(new Error(error.response));
+          setFetchRegister(false);
           showMessage({
             message: 'Cannot connect to server, please try again later',
             type: 'danger',
             backgroundColor: colors.red
           });
-        }
-      })
-      .catch((error) => {
-        crashlytics().recordError(new Error(error.response));
-        setFetchRegister(false);
-        showMessage({
-          message: 'Cannot connect to server, please try again later',
-          type: 'danger',
-          backgroundColor: colors.red
         });
+    } catch (e) {
+      crashlytics().recordError(new Error(e));
+      setFetchRegister(false);
+      showMessage({
+        message: 'Cannot connect to server, please try again later',
+        type: 'danger',
+        backgroundColor: colors.red
       });
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The updated code has improved error handling by wrapping the entire registration process in a try-catch block. However, there's a potential issue with the way errors are being handled. The same error message 'Cannot connect to server, please try again later' is used for all types of errors which might not be accurate or helpful for debugging. It would be better to display or at least log the actual error message.

```diff
- crashlytics().recordError(new Error(e));
- showMessage({
-   message: 'Cannot connect to server, please try again later',
-   type: 'danger',
-   backgroundColor: colors.red
- });

+ crashlytics().recordError(new Error(e));
+ showMessage({
+   message: e.message || 'Cannot connect to server, please try again later',
+   type: 'danger',
+   backgroundColor: colors.red
+ });
```

This change should be applied to lines 205-206, 218-223, 227-233, and 236-242. This will provide more detailed error messages which can help in identifying the root cause of any issues that may arise.


<!-- This is an auto-generated comment by OSS CodeRabbit -->

Reply:
We keep this because we need to passing `e.message || 'Cannot connect to server, please try again later'`